### PR TITLE
feature(locksmith): Multiple signers

### PR DESCRIPTION
**Pull request by julien51**
# Description

As we're working on using 3rd party APIs (like #12002 ) to send transactions, I realized that a good prerequisite work would be to add support to having multiple signers work at the same time in locksmith (right now our code assumes there is just one).

This means doing several changes:
* adding support for multiple signers to the hooks that look at signers
  * Captcha
  * Guild
  In both cases, we look up the lock's hook contract and check if any of the current signers is a signer on the hook and then use that one.
* Adding support for multiple signers to the step to purchase by card
  * for each of the current signer, we check if they are key granters. If they are, we use that one. Otherwise we show an error.

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/src/config/config.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ const config = {
   recaptchaSecret: process.env.RECAPTCHA_SECRET,
   logtailSourceToken: process.env.LOGTAIL,
   sessionDuration: Number(process.env.SESSION_DURATION || 86400 * 60), // 60 days
-  requestTimeout: '10s',
```
Comment (julien51) [inline locksmith/src/config/config.ts:87]: This was different in our 2 configs... restoring 25s for consistency.

File: locksmith/src/controllers/purchaseController.ts [hunk 2]
```diff
@@ -72,17 +72,18 @@ export class PurchaseController {
       purchaseType,
     } = await PaymentCaptureBody.parseAsync(request.body)
     const dispatcher = new Dispatcher()
-    const hasEnoughToPayForGas = await dispatcher.hasFundsForTransaction(
-      network
-    )
+
+    const [hasEnoughToPayForGas, soldOut] = await Promise.all([
```
Comment (julien51) [inline locksmith/src/controllers/purchaseController.ts:76]: Parallelizing to make the response snappier.

**Comment by julien51** [comment]
Once this is merged I will do extensive testing on staging to make sure we have everything (still) working as expected..

**Review by clemsos — APPROVED** [review]
LG 👍

**Pull request merged**